product_dimension/models/product.py

```diff
@@ -3,23 +3,24 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 
-from odoo import models, fields
-from odoo import api
+from odoo import models, fields, api
 
 
 class Product(models.Model):
     _inherit = 'product.product'
 
     @api.onchange('length', 'height', 'width', 'dimensional_uom_id')
     def onchange_calculate_volume(self):
-        if (not self.length or not self.height or not self.width or
-                not self.dimensional_uom_id):
-            return False
-
-        length_m = self.convert_to_meters(self.length, self.dimensional_uom_id)
-        height_m = self.convert_to_meters(self.height, self.dimensional_uom_id)
-        width_m = self.convert_to_meters(self.width, self.dimensional_uom_id)
-        self.volume = length_m * height_m * width_m
+        if self.length and self.height and self.width and \
+                self.dimensional_uom_id:
+            length_m = self.convert_to_meters(
+                self.length, self.dimensional_uom_id)
+            height_m = self.convert_to_meters(
+                self.height, self.dimensional_uom_id)
+            width_m = self.convert_to_meters(
+                self.width, self.dimensional_uom_id)
+            self.volume = length_m * height_m * width_m
+        return False
```

Comment:
With this change, the function will always return ``False``. Is it really the wanted behaviour?
Wasn't the function already OK?

Reply:
OK correct. I changed to previous behavior. Function was re-written to increase code coverage